include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -631,6 +631,12 @@ namespace VectorTools
         if (!cell1->active() && !cell2->active())
           continue;
 
+        // skip foreign cells
+        if (cell1->active() && !cell1->is_locally_owned())
+          continue;
+        if (cell2->active() && !cell2->is_locally_owned())
+          continue;
```

Comment:
The check needs to be that if one cell is locally owned, the other cell or all of it's children need to be locally owned.